# HG changeset patch
# User Rik <rik@octave.org>
# Date 1388691103 28800
#      Thu Jan 02 11:31:43 2014 -0800
# Node ID d3c6842ef18894ae83cb92b4e35dc2f3352f2bdc
# Parent  31d8e19a745dbdc9895f892261fd72e5a7a580ff
mex.m: Return status if called with an output argument (bug #40984).

* mex.m: Return status if called with an output argument (bug #40984).

diff --git a/scripts/miscellaneous/mex.m b/scripts/miscellaneous/mex.m
--- a/scripts/miscellaneous/mex.m
+++ b/scripts/miscellaneous/mex.m
@@ -18,13 +18,21 @@
 
 ## -*- texinfo -*-
 ## @deftypefn {Command} {} mex [options] file @dots{}
 ## Compile source code written in C, C++, or Fortran, to a MEX file.
 ## This is equivalent to @code{mkoctfile --mex [options] file}.
 ## @seealso{mkoctfile}
 ## @end deftypefn
 
-function mex (varargin)
-  args = {"--mex", varargin{:}};
-  mkoctfile (args{:});
+function retval = mex (varargin)
+
+  [output, status] = mkoctfile ("--mex", varargin{:});
+
+  if (! isempty (output))
+    disp (output);
+  endif
+  if (nargout > 0)
+    retval = status;
+  endif
+
 endfunction
 
